src/filebuf.c

```diff
@@ -71,6 +71,21 @@ static int lock_file(git_filebuf *file, int flags, mode_t mode)
 		char buffer[FILEIO_BUFSIZE];
 		ssize_t read_bytes;
 
+#ifdef GIT_WIN32
+		bool hidden = false;
+		if (strstr(file->path_original, ".git") == NULL) {
```

Comment:
I don't think that you want to use `strstr` here, by the way.  That will find a `.git` anywhere in the string, no?  You actually only care about the `.git` directory, which will take a more nuanced approach to consider.